File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        required_area = 0
        left_side = 0
        right_side = len(height) - 1
        difference = right_side - left_side
        while difference > 0:
            required_area = max(required_area, min(height[left_side], height[right_side]) * difference)
            if (height[left_side] <= height[right_side]):
                left_side += 1
            elif (height[left_side] > height[right_side]):
                right_side -= 1
            difference = right_side - left_side
        
        return required_area
            

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var total_area = 0
    var left_starter = 0
    var right_starter = height.length - 1
    var difference = right_starter - left_starter

    while (difference > 0) {
        total_area = Math.max(total_area, Math.min(height[left_starter], height[right_starter]) * difference)
        if (height[left_starter] <= height[right_starter])
            left_starter += 1
        else if (height[left_starter] > height[right_starter])
            right_starter -= 1
        difference = right_starter - left_starter
    }

    return total_area
};
